tools/system_libs.py

```diff
@@ -823,7 +822,7 @@ def get_files(self):
         filenames=['vfprintf.c'])
 
   def can_build(self):
-    return super(libprintf_long_double, self).can_build() and shared.Settings.WASM_BACKEND
+    return super(libprintf_long_double, self).can_build() and shared.Settings.WASM_BACKEND and shared.Settings.PRINTF_LONG_DOUBLE
```

Comment:
this doesn't match my understanding of `can_build` (i think this should go under `can_use` instead). but maybe I have it wrong? why was this change needed?